### PR TITLE
Team1

### PR DESCRIPTION
**Pull request by happybridge14-totoro**
Team1&4's MVC frame and demo codes.

### CODE REVIEW

File: src/m1/team14/controller/AbstractController.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            try {
+                Method method = model.getClass().
+                    getMethod("setTotalNumber", new Class[] {
+                                                      newValue.getClass()
+                                                  }
+                             );
+                method.invoke(model, newValue);
+            } catch (Exception ex) {
+              System.out.println("error?");
+              System.out.println(ex);
```
Comment (bates-hu) [inline src/m1/team14/controller/AbstractController.java]: use ex.printStackTrace()

Comment (happybridge14-totoro) [reply src/m1/team14/controller/AbstractController.java]: Thanks, will update the error handling functions.

File: src/m1/team14/controller/AbstractController.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     *
+     * @param propertyName = The name of the property.
+     * @param newValue = An object that represents the new value
+     * of the property.
+     */
+    protected void setModelProperty(String propertyName, Object newValue) {
+        for (AbstractModel model: registeredModels) {
+            try {
+                Method method = model.getClass().
+                    getMethod("setTotalNumber", new Class[] {
```
Comment (BrunoHu) [inline src/m1/team14/controller/AbstractController.java]: I am a little confused why a magic string appears in the abstract class

File: src/m1/team14/controller/AbstractController.java (hunk 1 above)
Comment (BrunoHu) [inline src/m1/team14/controller/AbstractController.java]: use ex.printStackTracc()

**Comment by happybridge14-totoro** [comment]
Update error tracking methods according to @bates-hu's comments.

File: src/m1/team14/view/AbstractViewPanel.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package m1.team14.view;
```
Comment (BrunoHu) [inline src/m1/team14/view/AbstractViewPanel.java]: commonly,
the name of Interface is like IViewPanel
the name of Abstract class is like AbstractViewPanel

**Review by BrunoHu — APPROVED** [review]

**Pull request merged**